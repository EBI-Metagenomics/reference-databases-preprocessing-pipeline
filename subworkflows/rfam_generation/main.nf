
include { UNCOMPRESS_FILE } from '../../modules/local/uncompress_file/main.nf'
include { EXTRACT_RFAM_SUBSET } from '../../modules/local/extract_rfam_subset/main.nf'

workflow RFAM_GENERATION {

    main:

        rfam_claninfo = file(params.rfam_claninfo, checkIfExists: true)
        rfam_cm = file(params.rfam_download_cm, checkIfExists: true)

        UNCOMPRESS_FILE(
            rfam_cm,
            "full_ribo.cm"
        )

        EXTRACT_RFAM_SUBSET(
            UNCOMPRESS_FILE.out.uncmp_file,
            rfam_claninfo,
            params.rfam_version,
            params.rfam_label
        )

    emit:
        ribo_cm = EXTRACT_RFAM_SUBSET.out.ribo_cm
        ribo_claninfo = EXTRACT_RFAM_SUBSET.out.ribo_claninfo
}