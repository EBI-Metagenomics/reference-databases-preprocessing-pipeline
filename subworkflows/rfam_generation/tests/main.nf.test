nextflow_workflow {

    name "Test Workflow RFAM_GENERATION"
    script "subworkflows/rfam_generation/main.nf"
    workflow "RFAM_GENERATION"
    tag "subworkflows"
    tag "rfam_generation"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
                rfam_version = "14.10"
                rfam_label = "RFAM"
                rfam_claninfo = "${projectDir}/subworkflows/rfam_generation/tests/data/temp_ribo.claninfo"
                rfam_download_cm = "${projectDir}/subworkflows/rfam_generation/tests/data/Rfam.cm.gz"
            }

        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.ribo_cm).match("ribo_cm") },
                { assert snapshot(workflow.out.ribo_claninfo).match("ribo_claninfo") }
            )
        }

    }

}
