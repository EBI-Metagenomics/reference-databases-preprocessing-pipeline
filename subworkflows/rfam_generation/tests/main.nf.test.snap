{
    "ribo_claninfo": {
        "content": [
            [
                "ribo.clan_info:md5,ed55b0130da268e5deca604fa34da618"
            ]
        ],
        "timestamp": "2023-12-13T11:32:56.609019052"
    },
    "ribo_cm": {
        "content": [
            [
                "ribo.cm:md5,fd8a38177a9ebefd89029cab2038edb7"
            ]
        ],
        "timestamp": "2023-12-13T11:32:56.443459095"
    }
}